vsphere/datadog_checks/vsphere/utils.py

```diff
@@ -0,0 +1,137 @@
+# (C) Datadog, Inc. 2019-present
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+from pyVmomi import vim
+
+from datadog_checks.base import ensure_unicode
+from datadog_checks.vsphere.constants import MOR_TYPE_AS_STRING, SHORT_ROLLUP
+
+
+def format_metric_name(counter):
+    return "{}.{}.{}".format(
+        ensure_unicode(counter.groupInfo.key),
+        ensure_unicode(counter.nameInfo.key),
+        ensure_unicode(SHORT_ROLLUP[str(counter.rollupType)]),
+    )
+
+
+def match_any_regex(string, regexes):
+    for regex in regexes:
+        match = regex.match(string)
+        if match:
+            return True
+    return False
+
+
+def is_resource_excluded_by_filters(mor, infrastructure_data, resource_filters):
+    resource_type = MOR_TYPE_AS_STRING[type(mor)]
+
+    for f in resource_filters:
+        if f[0] == resource_type:
```

Comment:
I'm not sure to understand. Is `resource_filters` a `blacklist` ?

The name `resource_filters` seems to suggest it's a whitelist.

Reply:
It's a whitelist indeed.
The function is called `is_resource_excluded_by_filters` and it returns False if any of the filters matches.

This line you've quoted is just making sure that when there is no filter we collect everything.
Reverting the logic introduced in https://github.com/DataDog/integrations-core/pull/5251#discussion_r363394016 by the way

Reply:
If the filter is a whitelist, reversing the name may make it more readable `is_resource_excluded_by_filters` => e.g. `is_monitored_resource`